lib/chef/cookbook_site_streaming_uploader.rb

```diff
@@ -1,244 +0,0 @@
-#
-# Author:: Stanislav Vitvitskiy
-# Author:: Nuo Yan (nuo@chef.io)
-# Author:: Christopher Walters (<cw@chef.io>)
-# Copyright:: Copyright (c) Chef Software Inc.
-# License:: Apache License, Version 2.0
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-
-autoload :URI, "uri"
-module Net
-  autoload :HTTP, "net/http"
-end
-autoload :OpenSSL, "openssl"
-module Mixlib
-  module Authentication
-    autoload :SignedHeaderAuth, "mixlib/authentication/signedheaderauth"
-  end
-end
-require "chef-utils/dist" unless defined?(ChefUtils::Dist)
-
-class Chef
-  # == Chef::CookbookSiteStreamingUploader
-  # A streaming multipart HTTP upload implementation. Used to upload cookbooks
-  # (in tarball form) to https://supermarket.chef.io
-  #
-  # inspired by http://stanislavvitvitskiy.blogspot.com/2008/12/multipart-post-in-ruby.html
-  class CookbookSiteStreamingUploader
```

Comment:
removing this probably breaks berkshelf?  it might be easier to leave this here to avoid berks having to take a dep on knife.